### PR TITLE
added super-resolution example

### PR DESCRIPTION
**Pull request by VictorIyke**
Added super-resolution example using ORT that could be run on both Android and Web

### CODE REVIEW

File: js/app-super-resolution/.gitignore [hunk 1]
```diff
@@ -0,0 +1,75 @@
+node_modules/
+.expo/
```
Comment (natke) [inline js/app-super-resolution/.gitignore:2]: We should add .expo_shared here too (I think) and then remove the files in .expo_shared from the repo

File: js/app-super-resolution/README.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# RUNNING SUPER-RESOLUTION ON IMAGES ACROSS VARIOUS PLATFORMS USING ONNX RUNTIME MODELS
```
Comment (natke) [inline js/app-super-resolution/README.md:1]: Let's make it clear here that the app runs on web and android and not iOS yet.

And if you know, the steps needed to get it working on iOS

File: js/app-super-resolution/README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+       ```sh
+       npm install -g yarn
+       ```
+       
+1. Setup project
+    1. Create Root Directory
+    2. Copy contents of this Repo into created directory
+
+    **NOTE:**
+    - `<SOURCE_ROOT>` refers to the root folder of the source code directory.
```
Comment (natke) [inline js/app-super-resolution/README.md:28]: Is this the root of the onnxruntime-inference-examples, or the app directory?

Do they need to copy, or can the clone the examples repo and change directories into the app folder?

File: js/app-super-resolution/README.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+ 
+2. Install required Libraries
+   1. Run the following command in `<SOURCE_ROOT>` to install the modules and libraries specified in the package.json file.
+      1. Using YARN:
+         ```sh
+         yarn
+         ```
+      
+      2. Using NPM:
```
Comment (natke) [inline js/app-super-resolution/README.md:37]: Do we need to do this if we are using expo?

**Review by natke — COMMENTED** [review]
Looking good. A couple of comments:
- which of the files in the android folder are actually source files? We should remove any files that get built?
- would there be any way of merging the screens for the mobile and web app more, so that more code is cross platform?

File: js/app-super-resolution/README.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+     **NOTE:**
+     If you run into dependency issues, you can run this instead to install the modules.
+      ```sh
+      npm install --force
+      ```
+   
+3. Run the following commands to launch:
+
+    In `<SOURCE_ROOT>`, run the following command to launch for Android
+    ```sh
```
Comment (YUNQIUGUO) [inline js/app-super-resolution/README.md:52]: nit: the commands for android/web is meant to be different? 

Is there a way to make it more consistent?

File: js/app-super-resolution/README.md (hunk 2 above)
Comment (YUNQIUGUO) [inline js/app-super-resolution/README.md:1]: general suggestion for read.me: 

maybe can make it more complete follow the same pattern/sections from maui example:

https://github.com/microsoft/onnxruntime-inference-examples/blob/main/mobile/examples/Maui/MauiVisionSample/readme.md

maybe for sections like"pre/post processing" methods, can expand to explain in more details in github tutorial page, and briefly talk about it here would be good.

File: js/app-super-resolution/screens/mobile.tsx [hunk 6]
```diff
@@ -0,0 +1,251 @@
+import { StatusBar } from 'expo-status-bar';
```
Comment (YUNQIUGUO) [inline js/app-super-resolution/screens/mobile.tsx:1]: general comments:
 
Microsoft license required at the beginning of each file. (can see other files in the repo as an example.)

also maybe the function name itself is sometimes self-explanatory, still can benefit from adding some comments especially for some of the main source files. briefly explain what each function does.

File: js/app-super-resolution/README.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+2. WEB
+  <img width="1215" alt="image" src="https://user-images.githubusercontent.com/106185642/181638855-f341e52e-dfc1-4362-b93a-0117f0cfd65a.png">
+   
+
+## Steps of building project from scratch
+0. Prerequisites
+    1. Install Node.js
+    2. Install Java and OpenJDK
+    3. Install expo
```
Comment (YUNQIUGUO) [inline js/app-super-resolution/README.md:14]: node.js/java, etc. maybe common for people,

but for expo fairly new concept, can add link to the name, for people to conveniently click and check. also doesn't hurt to provide links for openjdk, etc. ( for example, is there a specific version required to install?)

File: js/app-super-resolution/screens/mobile.tsx [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+    ])
+
+    setOutputImage({ localUri: imageRotated.uri })
+    return imageUri
+  };
+
+
+  async function loadModel() {
+    try {
+      const assets = await Asset.loadAsync(require('../assets/super_resnet12.ort'));
```
Comment (YUNQIUGUO) [inline js/app-super-resolution/screens/mobile.tsx:160]: `../assets/super_resnet12.ort` 

How is the model introduced and used in the example? is it loaded from the assets folder?  (um didn't see it in the folder.)

also in tutorial/readme.md file, would be good to mention models used in this example, limitations about it,  other models can try in the future, etc.

File: js/app-super-resolution/screens/mobile.tsx [hunk 9]
```diff
@@ -0,0 +1,251 @@
...
+        </View>
+      }
+
+
+      <StatusBar style="auto" />
+    </View>
+  );
+};
+
+
```
Comment (YUNQIUGUO) [inline js/app-super-resolution/screens/mobile.tsx:248]: nit: remove extra space/lines here

**Comment by fs-eire** [comment]
I have an important question: is this expo project using bare workflow or managed workflow? In other words, after you use the expo-cli to generate the `android` folder, did you do any changes to it?

If no, then you are using managed workflow. The whole `android` folder should not be included in the code base because they are generated.

File: js/app-super-resolution/.gitignore (hunk 1 above)
Comment (VictorIyke) [reply js/app-super-resolution/.gitignore:2]: 👍

File: js/app-super-resolution/README.md (hunk 2 above)
Comment (VictorIyke) [reply js/app-super-resolution/README.md:5]: 👍

File: js/app-super-resolution/README.md (hunk 3 above)
Comment (VictorIyke) [reply js/app-super-resolution/README.md]: It refers to the app directory. They can clone the examples repo and change directories into the app folder. I will specify that in the README

File: js/app-super-resolution/README.md (hunk 4 above)
Comment (VictorIyke) [reply js/app-super-resolution/README.md]: You can do either. I will specify that

File: js/app-super-resolution/README.md (hunk 5 above)
Comment (VictorIyke) [reply js/app-super-resolution/README.md]: From my knowledge, it should be different as Expo takes some extra steps in order to build the app for Android (also the same for IOS)

File: js/app-super-resolution/README.md (hunk 2 above)
Comment (VictorIyke) [reply js/app-super-resolution/README.md:5]: adding that now

File: js/app-super-resolution/screens/mobile.tsx (hunk 6 above)
Comment (VictorIyke) [reply js/app-super-resolution/screens/mobile.tsx:5]: Should the Microsoft license be at the beginning of every file?

File: js/app-super-resolution/README.md (hunk 7 above)
Comment (VictorIyke) [reply js/app-super-resolution/README.md]: sounds good, adding that

File: js/app-super-resolution/screens/mobile.tsx (hunk 8 above)
Comment (VictorIyke) [reply js/app-super-resolution/screens/mobile.tsx]: I just noticed that the model does not get pushed into the repo. So, I added a link to the model in the readme.md file

File: js/app-super-resolution/screens/mobile.tsx (hunk 9 above)
Comment (VictorIyke) [reply js/app-super-resolution/screens/mobile.tsx]: sure 👍

**Comment by VictorIyke** [comment]
@fs-eire 
To answer your question, I made a slight change to a file in the `android` folder and also added two files there. Although I have now removed the android folder from the code, I also explained, in the instructions.md file, how to use the expo-cli to generate the `android` folder and make the small changes.

File: js/app-super-resolution/instructions.md [hunk 10]
```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation.
+
+// Licensed under the MIT license.
```
Comment (fs-eire) [inline js/app-super-resolution/instructions.md:3]: the copyright banner should only be applied to source code files

Comment (VictorIyke) [reply js/app-super-resolution/instructions.md]: 👍 thanks

File: js/app-super-resolution/instructions.md [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+
+## Steps of building project from scratch
+
+0. Prerequisites
+
+   1. Install Node.js
+
+   2. Install Java (OpenJDK).
+
+      - Download and install [OpenJDK Version 11](https://adoptopenjdk.net/). Extract into the `program files` folder in your local disk.
```
Comment (fs-eire) [inline js/app-super-resolution/instructions.md:9]: extracting to any place should work (as long as the next step of setting JAVA_HOME is doing correctly).

Comment (VictorIyke) [reply js/app-super-resolution/instructions.md:9]: Oh, I did not know that. Thanks 👍

File: js/app-super-resolution/instructions.md [hunk 12]
```diff
@@ -0,0 +1,103 @@
+## Steps of building project from scratch
```
Comment (fs-eire) [inline js/app-super-resolution/instructions.md:1]: The instruction seems only work for Windows system. Is it expected?

Comment (VictorIyke) [reply js/app-super-resolution/instructions.md:1]: I believe it should only work for Windows. I am not really familiar with MacOS or Linux systems

**Comment by natke** [comment]
Superceded by #182

**Pull request closed without merging**